Add more usages for CN-faults (#493)
gmt-china/GMT_docs: source/dataset-CN/CN-faults.rst

CN-faults: 中国断层数据
=======================

CN-faults 数据来自邓起东院士的中国活动构造图(1:400万)，
由\ `国家地震科学数据中心——地震地质与地震动力学数据共享分中心 

> <http://datashare.igl.earthquake.cn/datashare>`__\ 提供。

Reviewer: 这个链接我打不开。此外，这个机构的名字是否可以只用**地震地质与地震动力学数据共享分中心 **？

Author: 最近这两天的确打不开，我正在问原因。如果只写**分中心**的话，别人会问是什么的分中心，不如只写一个**国家地震科学数据中心**

Reviewer: > 如果只写分中心的话，别人会问是什么的分中心，不如只写一个国家地震科学数据中心

okay.

Reviewer 2: 这个链接我可以正常打开。前几天打不开可能是因为七一网络管控。

包含了中国区域内主要断层的地理位置以及断层属性，
如断层名称、长度、走向、倾向、倾角、断层性质、最晚活动年代等。

数据文件
--------

> CN-faults 提供了一个数据文件\ :file:`CN-faults.gmt`\ ：中国区域主要断层

Reviewer: 因为我们约定了[文件风格](https://docs.gmt-china.org/latest/contributing/general-guidelines/#id6)，因此这里的第一个 `\` 不需要加:
> 4. 中文字符与英文字符和数字之间应加上空格，如 `中文 ABC 中文` 而非 `中文ABC中文`， `中文 123 中文` 而非 `中文123中文`

第二 `\` 需要加的原因是：
> 5. 标点符号采用全角，如 `，` `、`。``、``：``、``、``、``？` 等。标点符号与中文字符、英文字符以及数字之间不需加空格。
```suggestion
CN-faults 提供了一个数据文件 :file:`CN-faults.gmt`\ ：中国区域主要断层
```

CN-faults: 中国断层数据
=======================

CN-faults 数据来自邓起东院士的中国活动构造图(1:400万)，

> 由\ `国家地震科学数据中心——地震地质与地震动力学数据共享分中心 
> <http://datashare.igl.earthquake.cn/datashare>`__\ 提供。

Reviewer: ```suggestion
由\ `国家地震科学数据中心 <http://datashare.igl.earthquake.cn/datashare>`__\ 提供。
```

CN-faults: 中国断层数据
=======================

> CN-faults 数据来自邓起东院士的中国活动构造图(1:400万)，由 `国家地震科学数据中心——地震地质与地震动力学数据共享分中心 <http://datashare.igl.earthquake.cn/datashare>`__ 提供，包含了中国区域内主要断层的地理位置、断层名称、长度、走向、倾向、倾角、断层性质、最晚活动年代等信息。

Reviewer: ```suggestion
CN-faults 数据来自邓起东院士的中国活动构造图（1:400万），
由\ `国家地震科学数据中心—地震地质与地震动力学数据共享中心 <http://datashare.igl.earthquake.cn/datashare>`__\ 提供。
包含了中国区域内主要断层的地理位置以及断层属性，
如断层名称、长度、走向、倾向、倾角、断层性质、最晚活动年代等。
```

----

> CN-faults 数据来自邓起东院士的中国活动构造图(1:400万)，

Reviewer 2: ```suggestion
CN-faults 数据来自邓起东院士的中国活动构造图（1:400万），
```

由\ `国家地震科学数据中心 <http://datashare.igl.earthquake.cn/datashare>`__\ 提供。

> 包含了中国区域内主要断层的地理位置以及断层属性，
> 如断层名称、长度、走向、倾向、倾角、断层性质、最晚活动年代等。

Reviewer 2: ```suggestion
数据包含了中国区域内主要断层的地理位置以及断层属性，
如断层名称、长度、走向、倾向、倾角、断层性质、最晚活动年代等。
```

> 
> CN-faults 数据包含了中国区域内主要断层的地理位置信息。

Reviewer 2: ```suggestion
```

----

> CN-faults 数据来自于邓起东院士的中国活动构造图（1:400万），

Reviewer: ```suggestion
CN-faults 数据来自于邓起东院士编制的《中国活动构造图（1:400 万）》，
```

> CN-faults 数据来自于邓起东院士的中国活动构造图（1:400万），
> 由\ `国家地震科学数据中心 <http://datashare.igl.earthquake.cn/datashare>`__\ 提供原始数据。

Reviewer: ```suggestion
CN-faults 是\ `国家地震科学数据中心 <http://datashare.igl.earthquake.cn/datashare>`__\
提供的中国区域活动断裂数据，由邓起东院士编制的《中国活动构造图（1:400 万）》及相关资料经数字化处理获得。
```

Author: 这里我不建议添加“经数字化处理获得”，因为原本的《中国活动构造图（1:400 万）》图集就是屈老师负责制图的，所以她提供的矢量数据应该是一手资料，而不是二次数字化后的。以上也只是我的猜测，所以这里我们只提供数据出处即可。

Reviewer: 请看 http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html 第一段:

> 中国著名活动构造专家邓起东院士, 在多年潜心研究的基础上, 对中国几十年来的活动构造研究成果进行了详细的收集、整理和汇总, 并编制了最新版的1/400 万中国活动构造图。我所研究人员对这套图和相关的资料进行了数字化处理, 采用GIS技术和数据库结构模式, 建立了中国活动构造空间数据库, 实现了对这些宝贵资料科学有效的管理和灵活方便的应用, 使之能够更好地服务于社会。同时, 将不同时期、不同地域的资料集成在一起, 便于活动构造研究及其应用工作的进一步深化。

国家地震科学数据中心的**中国区域活动断层数据**本身是由这套图和相关的资料进行了数字化处理得到？

> CN-faults 数据来自邓起东院士的中国活动构造图（1:400万），
> 由\ `国家地震科学数据中心 <http://datashare.igl.earthquake.cn/datashare>`__\ 提供。
> 数据包含了中国区域内主要断层的地理位置以及断层属性，
> 如断层名称、长度、走向、倾向、倾角、断层性质、最晚活动年代等。

Reviewer 2: ```suggestion
CN-faults 数据来自于邓起东院士的中国活动构造图（1:400万），
由\ `国家地震科学数据中心 <http://datashare.igl.earthquake.cn/datashare>`__\ 提供原始数据。
数据包含了中国区域内主要断层的地理位置以及断层属性，
如断层名称、长度、走向、倾向、倾角、断层性质、最晚活动年代等。
```

标注断层名
++++++++++

> GMT中可以直接使用\ :doc:`/module/plot` 模块的 **-Sq** 选项标注断层名。

Reviewer: ```suggestion
GMT 中可以直接使用 :doc:`/module/plot` 模块的 **-Sq** 选项标注断层名。
```

标注断层名
++++++++++

> GMT 中可以直接使用 :doc:`/module/plot` 模块的 **-Sq** 选项标注断层名。

Reviewer: ```suggestion
可以使用 :doc:`/module/plot` 模块的 **-Sq** 选项标注断层名。
```

    gmt begin CN-faults-labeling png,pdf
        gmt coast -JM10c -RTW -Baf -W0.5p,black
        # 由于 -Sq 无法支持中文，该命令将断层名称输出到文件faultname.dat中，并绘制了一个全透明的图层
        # 实际上，当执行脚本获得faultname.dat 后，可将该命令注释掉
        gmt plot CN-faults.gmt -Sqn1:+Lh+tfaultname.dat -aL=断层名称 -t100
        # 使用 plot 绘制断层
        gmt plot CN-faults.gmt -W1p,red
        # 标注断层名
        gmt text faultname.dat -F+f15p,46,red+a

>         rm faultname.dat

Reviewer: 不绘制断层和断层名？

Author: 脚本中plot -Sq可同时绘制断层和断层名

>         # plot命令-Sq的+Lh子选项从当前段头中获取该标签值，作为该线段的文字标注

Reviewer: 请修改为英语注释

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-single-fault png
        # convert -S根据属性筛选某一条/类断层数据 

>         gmt convert CN-faults.gmt -S"断层名称=阿巴宫断裂"  -aL="断层名称" | gmt plot -R87/90/46/49 -Ba -Sqn1:+Lh+f11,STHeiti-Regular--GB-EUC-H 

Reviewer: 是否可以写为两行，一行太长了。

Author: 我想有三个解决方案：
1.  筛选结果输出为文件，把gmt convert和gmt plot 拆成两句；
2. 增加basemap，把plot的-R和-B写到前面。而标注文字的字体“STHeiti-Regular--GB-EUC-H”这个不建议改，因为不同用户的字体编号可能不同，这样写更保险一些；
3. 不举 **断层名称** 的例子，因为涉及到长长的中文名称

Reviewer 2: > 2\. 而标注文字的字体“STHeiti-Regular--GB-EUC-H”这个不建议改，因为不同用户的字体编号可能不同，这样写更保险一些；

中文的问题，我建议还是用字体编号表示。主要原因在于，`STHeiti-Regular--GB-EUC-H` 只适用于Windows，不适用于 Linux 或 macOS。而且，对于大部分用户而言，按照中文手册操作，中文字体的编号其实都是统一的了。

    gmt end show

根据空间范围提取数据
++++++++++++++++++++++

.. note::

>     :doc:`/module/gmtselect`模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。此处仅举其中一个例子。

Reviewer: ```suggestion
    :doc:`/module/gmtselect` 模块可以实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。
    此处仅举其中一个例子。
```

> .. note::
> 
>     :doc:`/module/gmtselect`模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。此处仅举其中一个例子。

Reviewer: ```suggestion
:doc:`/module/gmtselect` 模块可以实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。
此处仅举其中一个例子。
```

Author: 这里不需要缩进吗？ 这些缩进啊、编码啊、转义符啊真是麻烦死了

Reviewer: 删除了 `note` 指令就不要缩进了。把这些麻烦解决了，才能让用户不麻烦 :smile:

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-buffer-fault png
        # 圆心
        echo 111 35.5 >center
        # 半径100km的圆域
        echo 111 35.5 200k| gmt plot -R109/113/34/37 -JM15c -SE- -Wblue -fg
        # 筛选半径为100km内的断层

>         gmt select CN-faults.gmt -Ccenter+d100k -fg -aL="断层名称" | gmt plot -Ba -Sqd3c:+Lh+f11,STHeiti-Regular--GB-EUC-H

Reviewer: 是否可以写为两行，一行太长了。

        rm center
    gmt end show

数据来源及处理
--------------

> 1.  从 `中国大陆1：400万活动断层数据库 <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__ 下载 ``Active_fault.zip`` 压缩包

Reviewer: ```suggestion
1.  从 `中国大陆1：400万活动断层数据库 <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__
     下载 :file:`Active_fault.zip` 压缩包
```

Author: 我理解:file: 的意思是可以在页面下载的文件？还是说仅仅是一种展示样式？

Author: 我们需要将数据源也上传吗？

Reviewer: > 我理解:file: 的意思是可以在页面下载的文件？还是说仅仅是一种展示样式？

`:file:` 是一种语义标记，表示这是一个文件。类似的还有 `:kbd:` （表示是键盘键），`:download:` （表示可下载）等。他们的显示效果如下：

![](https://user-images.githubusercontent.com/50591376/123925500-8cd8d900-d9bd-11eb-9f4f-23807050bbec.png)

可以参考：
- https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html?highlight=%3Afile%3A#cross-referencing-syntax
- https://seismo-learn.org/contributing/restructuredtext/#id14

> 我们需要将数据源也上传吗？

暂时不需要，#48 提交的数据后面可以直接被缓存到 GitHub。手册中直接用文件名即可。

Author: 你refer错了，是geodataset的issue48 不是docs的

Author: > 

这里的超链接没有用 \， 为什么第一段加上了 \ 呢？

Reviewer: > 这里的超链接没有用 \， 为什么第一段加上了 \ 呢？

也应该加 `\`。审稿一般要好多轮，每一轮只是解决一部分问题。

加 `\` 的效果：
```
由\ `国家地震科学数据中心——地震地质与地震动力学数据共享分中心 
<http://datashare.igl.earthquake.cn/datashare>`__\ 提供。
```
![1](https://user-images.githubusercontent.com/50591376/124086972-25399100-da84-11eb-8266-74c98250a314.png)

不加 `\` 的效果：
```
由 `国家地震科学数据中心——地震地质与地震动力学数据共享分中心 
<http://datashare.igl.earthquake.cn/datashare>`__ 提供。
```
![2](https://user-images.githubusercontent.com/50591376/124086993-2b2f7200-da84-11eb-9db1-e617e7d902fe.png)

加 `\` 可以去掉空格。可以使用 http://rst.ninjs.org/ 在线快速测试，也可以在本地编译预览测试差异。

> 

Reviewer: ```suggestion
```

> 2. 使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式（若不关注属性信息，GMT也可以直接读取Shapefile格式源数据绘图）::

Reviewer: ```suggestion
2.  使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式
    （若不关注属性信息，GMT 也可以直接读取 Shapefile 格式的源数据来绘图）\ ::
```

Author: 什么时候断行用空格，什么时候不用？

Author: 为什么是 \:: 而不是 ::  这个斜杠是转义符吗？

Reviewer: 也可以可以不用 `\`. 涉及到中文和 Sphinx 语法符号时候是挺麻烦的。

    ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp --config SHAPE_ENCODING "UTF-8"

> 社区最终提供的:file:`CN-faults.gmt`\参考了《最新1/400万中国活动构造空间数据库的建立》（屈春燕，2008）对属性字段名称的缺失和错误部分进行了订正。

Reviewer: ```suggestion
社区最终提供的 :file:`CN-faults.gmt`\ 参考了
《最新1/400万中国活动构造空间数据库的建立》（屈春燕，2008）
对属性字段名称的缺失和错误部分进行了订正。
```

> 数据引用

Reviewer: 或许我们可以统一一下引用格式的书写 (#465). 比如，年份要不要加括号，[M], [J], [D] 这些要不要加。

--------

> - 中文引用1:: 邓起东. 中国活动构造图(1:400万)(附光盘)[M]. 地震出版社, 2007.

Reviewer: See https://item.jd.com/10514857.html
```suggestion
- 邓起东. 2007.
  中国活动构造图（1：400万）.
  地震出版社.
```

> - 中文引用2:: 屈春燕. 最新1/400万中国活动构造空间数据库的建立[J]. 地震地质, 2008, 30(1):298-304.

Reviewer: See http://dx.chinadoi.cn/10.3969/j.issn.0253-4967.2008.01.022
```suggestion
- 屈春燕. 2008.
  最新 1/400 万中国活动构造空间数据库的建立.
  地震地质, 30(1), 298-304.
  http://dx.chinadoi.cn/10.3969/j.issn.0253-4967.2008.01.022
```

Author: 为什么拆解成了这么多行？

Reviewer: 这样的格式维护者看着方便，网页显示效果是相同的。

@seisman 和我在维护 https://seismo-learn.org/software/fk/introduction/#id5 中，统一采用这种写法：

源码：
![](https://user-images.githubusercontent.com/50591376/123924169-446ceb80-d9bc-11eb-91eb-35ea04ca6735.png)

网页效果：
![](https://user-images.githubusercontent.com/50591376/123924420-885ff080-d9bc-11eb-8d3c-b11c41df5f27.png)

> - 英文引用:: Deng, Q. D., Zhang, P. Z., Ran, Y. K., Yang, X. P., Min, W., Chen, L. C., 2003. Active tectonics and earthquake activities in China. Earth Science Frontiers, 10(S1): 66-73.

Reviewer: ```suggestion
- Deng, Q. D., Zhang, P. Z., Ran, Y. K., Yang, X. P., Min, W., Chen, L. C., 2003.
  Active tectonics and earthquake activities in China.
  Earth Science Frontiers, 10(S1): 66-73.
  https://en.cnki.com.cn/Article_en/CJFDTotal-DXQY2003S1011.htm.
```

    gmt begin CN-faults-labeling png,pdf
        gmt coast -JM10c -RTW -Baf -W0.5p,black
        # -aL: extract "断层名称" as "-L" value in segment headers
        # -Sq+Lh: set the label text from "-L" in segment headers        
        gmt convert CN-faults.gmt -aL="断层名称" | gmt plot -Sqn1:+Lh+f11,39
    gmt end show

根据属性信息提取数据
++++++++++++++++++++++

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-single-fault png
        gmt basemap -R98/105/22/27 -Ba 
        # -S: output record contains specified field attribute
        gmt convert CN-faults.gmt -S"断层名称=红河断裂" | gmt plot
    gmt end show

根据空间范围提取数据
++++++++++++++++++++++

> :doc:`/module/gmtselect`\ 模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。

Reviewer: ```suggestion
:doc:`/module/gmtselect` 模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。
```

> 此处仅举其中一个例子。

Reviewer: 圆形是缓冲区例子吗？
```suggestion
此处仅以缓冲区范围为例。
```

Author: 是的，**点** 的缓冲区。

.. gmtplot::
   :show-code: true
   :width: 50%
   

>     gmt begin CN-buffer-fault png

Reviewer: 见[示例风格](https://docs.gmt-china.org/latest/contributing/general-guidelines/#id7):
> 所有脚本至少需要生成 PNG 格式的图片，建议使用 png,pdf 生成两种格式的图片，别供网页版和 PDF 版使用

```suggestion
    gmt begin CN-buffer-fault png,pdf
```

        gmt basemap -R109/113/34/37 -JM15c -Ba
        echo 111 35.5 >center
        # draw a cirle with a radius of 100km
        echo 111 35.5 200k| gmt plot -SE- -Wblue -fg
        # extracing faults in the circle
        gmt select CN-faults.gmt -Ccenter+d100k -fg | gmt plot
        rm center
    gmt end show

数据来源及处理
--------------

> 1. 从\ `中国大陆1：400万活动断层数据库 
> <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__\ 
> 下载\ :file:`Active_fault.zip`\ 压缩包

Reviewer: ```suggestion
1. 从\ `中国大陆1：400万活动断层数据库 <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__\
   下载 :file:`Active_fault.zip` 压缩包
```

Author: **下载**前面的缩进不会产生句子中的空格吗？  斜杠可以把N个空格都删除吗？

Reviewer: 下载前面的缩进跟斜杠没有关系。前面的缩进是因为这是个列表，需要跟上一行 (`1. 从...`) 对齐，也就是列表的语法。

请看 https://seismo-learn.org/contributing/restructuredtext/#id9 的 **列表** 例子。

> 2. 使用\ :doc:`ogr2ogr </table/ogr2ogr>`\ 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式

Reviewer: ```suggestion
2. 使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式::
```

> （若不关注属性信息，GMT也可以直接读取Shapefile格式源数据绘图）\::
> 
>     ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp --config SHAPE_ENCODING "UTF-8"

Reviewer: ```suggestion

        ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp --config SHAPE_ENCODING "UTF-8"
    
    若不关注属性信息，GMT 也可以直接读取 Shapefile 格式源数据绘图。

```

> 社区最终提供的\ :file:`CN-faults.gmt`\ 参考了

Reviewer: ```suggestion
社区最终提供的 :file:`CN-faults.gmt` 参考了
```

    gmt begin CN-faults-labeling png,pdf
        gmt coast -JM10c -RTW -Baf -W0.5p,black
        # -aL: extract "断层名称" as "-L" value in segment headers
        # -Sq+Lh: set the label text from "-L" in segment headers        
        gmt convert CN-faults.gmt -aL="断层名称" | gmt plot -Sqn1:+Lh+f11,39
    gmt end show

根据属性信息提取数据
++++++++++++++++++++++

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-single-fault png
        gmt basemap -R98/105/22/27 -Ba 
        # -S: output record contains specified field attribute
        gmt convert CN-faults.gmt -S"断层名称=红河断裂" | gmt plot
    gmt end show

根据空间范围提取数据
++++++++++++++++++++++

> .. note::
> 

Reviewer: ```suggestion
```

 :doc:`/module/gmtselect`模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。
 此处仅举其中一个例子。

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-buffer-fault png
        gmt basemap -R109/113/34/37 -JM15c -Ba
        echo 111 35.5 >center
        # draw a cirle with a radius of 100km
        echo 111 35.5 200k| gmt plot -SE- -Wblue -fg
        # extracing faults in the circle
        gmt select CN-faults.gmt -Ccenter+d100k -fg | gmt plot
        rm center
    gmt end show

数据来源及处理
--------------

> 1. 从 `中国大陆1：400万活动断层数据库 

Reviewer: ```suggestion
1. 从\ `中国大陆 1：400 万活动断层数据库 
```

> <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__ 

Reviewer: ```suggestion
<http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__\
```

下载 :file:`Active_fault.zip` 压缩包

2. 使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式

> （若不关注属性信息，GMT也可以直接读取Shapefile格式源数据绘图）\::

Reviewer: ```suggestion
（若不关注属性信息，也可以用 GMT 直接读取 Shapefile 格式的源数据进行绘图）\ ::
```

    ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp --config SHAPE_ENCODING "UTF-8"

> 社区最终提供的 :file:`CN-faults.gmt`\ 参考了

Reviewer: ```suggestion
社区最终提供的 :file:`CN-faults.gmt` 参考了
```

    gmt begin CN-faults-labeling png,pdf
        gmt coast -JM10c -RTW -Baf -W0.5p,black

>         # -aL: extract "断层名称" as "-L" value in segment headers

Reviewer: 我做了如下修改，是否符合原意？

```suggestion
        # -aL="断层名称": set the "L" value (i.e., label) in segment headers using "断层名称"
```

>         # -Sq+Lh: set the label text from "-L" in segment headers        

Reviewer: ```suggestion
        # :+Lh: take the label text from the "L" value in the segment header
```

        gmt convert CN-faults.gmt -aL="断层名称" | gmt plot -Sqn1:+Lh+f11p,39
    gmt end show

> 利用convert命令根据属性信息提取数据
> +++++++++++++++++++++++++++++++++

Reviewer: ```suggestion
根据属性信息提取数据
++++++++++++++++++++
```

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-single-fault png
        gmt basemap -R98/105/22/27 -Ba 
        # -S: output record contains specified field attribute

>         gmt convert CN-faults.gmt -S"断层名称=红河断裂" | gmt plot

Author: 我认为这个例子中投影不重要，但如果在其他版本上不设置投影会报错的话，那还是加一个吧。

    gmt end show

根据空间范围提取数据
++++++++++++++++++++++

:doc:`/module/gmtselect` 模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。
此处仅以缓冲区范围为例。

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-buffer-fault png,pdf
        gmt basemap -R109/113/34/37 -JM15c -Ba

>         # draw a cirle with a radius of 100km

Reviewer: ```suggestion
        # draw a circle with a radius of 100 km
```

>         echo 111 35.5 200k| gmt plot -SE- -Wblue -fg

Reviewer: ```suggestion
        echo 111 35.5 200k | gmt plot -SE- -Wblue -fg
```

>         # extracing faults in the circle

Reviewer: ```suggestion
        # extract faults within the circle
```

        gmt select CN-faults.gmt -C111/35.5+d100k -fg | gmt plot
    gmt end show

数据来源与处理
--------------

> 1. 从\ `中国大陆1：400万活动断层数据库 <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__\
>    下载 :file:`Active_fault.zip` 压缩包

Reviewer: ```suggestion
1.  下载 :file:`Active_fault.zip` 压缩包：\ `中国大陆地区 1:400 万活动断层数据库
    <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__
```

> 2. 使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式::
> 
>         ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp
>     
>    若不关注属性信息，GMT 也可以直接读取 Shapefile 格式源数据绘图。

Reviewer: ```suggestion
2.  格式转换

    使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式::

        ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp
    
    若不关注属性信息，也可以直接使用 GMT 读取 Shapefile 格式的源数据进行绘图。
```

> 

Reviewer: ```suggestion
```

> 社区最终提供的 :file:`CN-faults.gmt` 参考了
> 《最新1/400万中国活动构造空间数据库的建立》（屈春燕，2008）
> 对属性字段名称的缺失和错误部分进行了订正。

Reviewer: ```suggestion
社区最终提供的 :file:`CN-faults.gmt` 文件参考了
《最新 1/400 万中国活动构造空间数据库的建立》（屈春燕，2008），
对部分属性字段名称的缺失和错误进行了订正。
```

    gmt begin CN-faults-labeling png,pdf
        gmt coast -JM10c -RTW -Baf -W0.5p,black
        # -aL: extract "断层名称" as "-L" value in segment headers
        # -Sq+Lh: set the label text from "-L" in segment headers        

>         gmt convert CN-faults.gmt -aL="断层名称" | gmt plot -Sqn1:+Lh+f11,39

Reviewer 2: ```suggestion 
        gmt convert CN-faults.gmt -aL="断层名称" | gmt plot -Sqn1:+Lh+f11p,39
```

Reviewer 2: 能否直接用

```
gmt plot CN-faults.gmt -Sqn1:+Lh+f11,39 -aL="断层名称"
```

Author: 我这里不能，这样提取出来并画出来的不是“断层名称”这一属性列，而是第一列。 不知道你们行不行 。

    gmt end show

根据属性信息提取数据
++++++++++++++++++++++

> 

Reviewer 2: 其实在这里可以加一句说明，类似于“可以使用 convert 的 -S 选项筛选指定断层的数据”。

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-single-fault png
        gmt basemap -R98/105/22/27 -Ba 
        # -S: output record contains specified field attribute
        gmt convert CN-faults.gmt -S"断层名称=红河断裂" | gmt plot
    gmt end show

根据空间范围提取数据
++++++++++++++++++++++

:doc:`/module/gmtselect` 模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。
此处仅以缓冲区范围为例。

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-buffer-fault png,pdf
        gmt basemap -R109/113/34/37 -JM15c -Ba

>         echo 111 35.5 >center
>         # draw a cirle with a radius of 100km
>         echo 111 35.5 200k| gmt plot -SE- -Wblue -fg
>         # extracing faults in the circle
>         gmt select CN-faults.gmt -Ccenter+d100k -fg | gmt plot

Reviewer 2: 感觉直接这样写更简洁。
```suggestion
        # draw a cirle with a radius of 100km
        echo 111 35.5 200k| gmt plot -SE- -Wblue -fg
        # extracing faults in the circle
        gmt select CN-faults.gmt -C111/35.5+d100k -fg | gmt plot
```

Author: 学到了

        rm center
    gmt end show

数据来源与处理
--------------

1. 从\ `中国大陆1：400万活动断层数据库 <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__\
   下载 :file:`Active_fault.zip` 压缩包

2. 使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式::

        ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp
    

>     若不关注属性信息，GMT 也可以直接读取 Shapefile 格式源数据绘图。

Reviewer 2: ```suggestion
   若不关注属性信息，GMT 也可以直接读取 Shapefile 格式源数据绘图。
```

可以使用 **convert** 模块的 **-S** 选项提取指定属性的数据。
例如，下例中提取出“断层名称”为“红河断裂”的断层数据。

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-single-fault png
        gmt basemap -R98/105/22/27 -JM15c -Ba 
        # -S: output record contains specified field attribute
        gmt convert CN-faults.gmt -S"断层名称=红河断裂" | gmt plot
    gmt end show

根据空间范围提取数据
++++++++++++++++++++++

:doc:`/module/gmtselect` 模块，可实现矩形范围、缓冲区范围和多边形范围内空间数据的提取。
此处仅以缓冲区范围为例。

.. gmtplot::
   :show-code: true
   :width: 50%
   
    gmt begin CN-buffer-fault png,pdf
        gmt basemap -R109/113/34/37 -JM15c -Ba
        # draw a circle with a radius of 100 km
        echo 111 35.5 200k | gmt plot -SE- -Wblue -fg
        # extract faults within the circle
        gmt select CN-faults.gmt -C111/35.5+d100k -fg | gmt plot
    gmt end show

数据来源与处理
--------------

1.  下载 :file:`Active_fault.zip` 压缩包：\ `中国大陆地区 1:400 万活动断层数据库
    <http://datashare.igl.earthquake.cn/map/ActiveFault/introFault.html>`__

2.  格式转换

    使用 :doc:`ogr2ogr </table/ogr2ogr>` 将 Shapefile 格式转换为 GMT 可识别的 OGR/GMT 格式::

        ogr2ogr -f GMT CN-faults.gmt 中国断层_邓起东Line_Project.shp
    
    若不关注属性信息，也可以直接使用 GMT 读取 Shapefile 格式的源数据进行绘图。

> 社区最终提供的 :file:`CN-faults.gmt` 文件参考了
> 《最新 1/400 万中国活动构造空间数据库的建立》（屈春燕，2008），
> 对部分属性字段名称的缺失和错误进行了订正。

Reviewer: ```suggestion
社区提供的 :file:`CN-faults.gmt` 文件还参考了《最新 1/400 万中国活动构造空间数据库的建立》，
对部分属性字段名称的缺失和错误进行了修正。
```

数据引用
--------

- 邓起东. 2007. 

>   中国活动构造图(1:400万). 

Reviewer: ```suggestion
  中国活动构造图 (1:400 万).
```

  地震出版社.
- 屈春燕. 2008.

>   最新1/400万中国活动构造空间数据库的建立.

Reviewer: ```suggestion
  最新 1/400 万中国活动构造空间数据库的建立.
```

>   地震地质,30(1):298-304.

Reviewer: ```suggestion
  地震地质, 30(1):298-304.
```
